llarp/path/path.hpp

```diff
@@ -127,17 +128,44 @@ namespace llarp
         }
       };
 
+      /// hash for std::shared_ptr
       struct Ptr_Hash
       {
         size_t
-        operator()(const std::shared_ptr< Path >& p) const
+        operator()(const Path_ptr& p) const
         {
           if(p == nullptr)
             return 0;
           return Hash{}(*p);
         }
       };
 
+      /// hash for std::shared_ptr by path endpoint
+      struct Endpoint_Hash
+      {
+        size_t
+        operator()(const Path_ptr& p) const
+        {
```

Comment:
Is a `std::function` or lambda not a valid parameter to `std::unordered_set`'s hash template parameter?

Reply:
no idea

Reply:
No, the template parameter needs to be a default-constructible type.